### PR TITLE
Added possibility to ignore rectors per statement

### PR DESCRIPTION
**Pull request by jeroensmit**
This would be a solution for issue #1591 

In my experience it can be useful to exclude a single statement. 

I took the liberty of writing the feature first and discuss it afterwards. 

So I look to know what you thing of this approach.

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Hi, I'm usually against these custom doc blocks in code. 
But in combination with other tool ignores, it could make sense.

To continue I need from you:

- Numbers on performance drop on checking every node's annotation
- Twitter or any other pool feedback from community (I'll retweet on @rectorphp)
- More tests, e.g. on FileSystemRector as well

File: src/Rector/AbstractRector.php [hunk 1]
```diff
@@ -174,6 +196,18 @@ protected function isAtLeastPhpVersion(string $version): bool
...
+    {
+        if (preg_match_all('#@noRector\s*(?<rectorName>\S+)#i', $doc->getText(), $matches)) {
+            foreach ($matches['rectorName'] as $ignoreSpec) {
+                if (static::class === ltrim($ignoreSpec, '\\')) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
```
Comment (TomasVotruba) [inline src/Rector/AbstractRector.php]: The logic should be decoupled to own service

**Comment by jeroensmit** [comment]
Came up with a different solution, which I think is better than what I did before...
Coming up soon ;)

**Comment by TomasVotruba** [comment]
Replaced by #2087

**Pull request closed without merging**